DatadogSessionReplay/Sources/Recorder/ViewTreeSnapshotProducer/ViewTreeSnapshot/NodeRecorders/UIHostingViewRecorder.swift

```diff
@@ -78,19 +85,19 @@ internal class UIHostingViewRecorder: NodeRecorder {
         let node = Node(viewAttributes: attributes, wireframesBuilder: builder)
         return SpecificElement(subtreeStrategy: .record, nodes: [node])
     }
-}
 
-@available(iOS 18.1, tvOS 18.1, *)
-internal class iOS18HostingViewRecorder: UIHostingViewRecorder {
-    override func semantics(reflecting subject: AnyObject, nodeID: NodeID, with attributes: ViewAttributes, in context: ViewTreeRecordingContext) throws -> NodeSemantics? {
-        guard
-            let ivar = class_getInstanceVariable(type(of: subject), "_base"),
-            let _base = object_getIvar(subject, ivar) as? AnyObject
-        else {
-            return nil
+    private func extractObject(from subject: AnyObject, keyPath: String) -> AnyObject? {
+        var current = subject
+        for component in keyPath.split(separator: ".") {
```

Comment:
suggestion/ Lets use an array as `keyPath` and avoid splitting the string at each recording phase. We could also use static `keyPaths` to avoid their allocations :)

Reply:
You are absolutely right. I'll fix it. Thanks!